django/views/decorators/debug.py

```diff
@@ -38,42 +38,43 @@ def my_function()
         )
 
     def decorator(func):
-        @wraps(func)
-        def sensitive_variables_wrapper(*func_args, **func_kwargs):
-            if variables:
-                sensitive_variables_wrapper.sensitive_variables = variables
-            else:
-                sensitive_variables_wrapper.sensitive_variables = "__ALL__"
-            return func(*func_args, **func_kwargs)
-
         if iscoroutinefunction(func):
-            markcoroutinefunction(sensitive_variables_wrapper)
+
+            @wraps(func)
+            async def sensitive_variables_wrapper(*func_args, **func_kwargs):
+                return await func(*func_args, **func_kwargs)
 
             wrapped_func = func
             while getattr(wrapped_func, "__wrapped__", None) is not None:
                 wrapped_func = wrapped_func.__wrapped__
 
             try:
                 file_path = inspect.getfile(wrapped_func)
-            except TypeError:  # raised for builtins
-                file_path = None
-
-            try:
-                _file_lines, first_file_line = inspect.getsourcelines(wrapped_func)
-            except TypeError:  # raised for builtins
-                first_file_line = None
-            assert file_path is not None and first_file_line is not None, (
-                f"{func.__name__} cannot safely be wrapped by "
-                + "@sensitive_variables_wrapper, please make it "
-                + "either non-async or defined in a file (and not a builtin)"
-            )
-            key = hash(f"{file_path}:{first_file_line}")
+                _, first_file_line = inspect.getsourcelines(wrapped_func)
+            except TypeError:  # Raises for builtins.
+                raise ValueError(
+                    f"{func.__name__} cannot safely be wrapped by "
+                    "@sensitive_variables, please make it either non-async or defined "
+                    "in a file (and not a builtin)."
+                )
```

Comment:
I wonder if we can cover this with a test somehow. I don't see any functions in CPython's Lib/asyncio folder that reference any C modules but I just skimmed it.

IMO we should change the wording slightly:

```
"please make it either non-async or defined in a Python file (not a builtin or from a native extension)"
``` 

This is a bit more wordy but might help users see what they need to do? Right now if I saw the current error message I might not be sure what to do to fix it :/

Reply:
> I wonder if we can cover this with a test somehow. I don't see any functions in CPython's Lib/asyncio folder that reference any C modules but I just skimmed it.

Unfortunately, neither do I :disappointed:

Reply:
I checked and there is nothing in the `asyncio` module, so I'm fine to leave it untested.